convert/harness/downgrader/downgrade.go

```diff
@@ -574,6 +579,70 @@ func (d *Downgrader) convertStepPlugin(src *v1.Step) *v0.Step {
 	}
 }
 
+// helper function to convert a Plugin step from the v1
+// structure to the v0 harness structure.
+//
+// TODO convert resources
+// TODO convert reports
+func (d *Downgrader) convertStepPluginToDocker(src *v1.Step) *v0.Step {
+	spec_ := src.Spec.(*v1.StepPlugin)
+	var id = d.identifiers.Generate(
+		slug.Create(src.Id),
+		slug.Create(src.Name),
+		slug.Create(src.Type))
+	if src.Name == "" {
+		src.Name = id
+	}
+
+	stepDocker := &v0.StepDocker{
+		Caching:      true,
+		ConnectorRef: d.dockerhubConn,
+		Privileged:   spec_.Privileged,
+		RunAsUser:    spec_.User,
+	}
+
+	// Check if "repo" exists in spec_.With
+	if repo, ok := spec_.With["repo"].(string); ok {
+		// If "repo" exists, set it in StepDocker
+		stepDocker.Repo = repo
+	}
+	if context, ok := spec_.With["context"].(string); ok {
+		// If "context" exists, set it in StepDocker
+		stepDocker.Context = context
+	}
+	if dockerfile, ok := spec_.With["dockerfile"].(string); ok {
+		// If "dockerfile" exists, set it in StepDocker
+		stepDocker.Dockerfile = dockerfile
+	}
+	if target, ok := spec_.With["target"].(string); ok {
+		// If "target" exists, set it in StepDocker
+		stepDocker.Target = target
+	}
+	if cacheRepo, ok := spec_.With["cache_repo"].(string); ok {
+		// If "cache_repo" exists, set it in StepDocker
+		stepDocker.RemoteCacheRepo = cacheRepo
```

Comment:
What's this field? Is this the registry caching field? For DLC we don't use this right because the OOTB step for Cloud infra will not have it

Reply:
Commented it out